zone/attack.cpp

```diff
@@ -2571,7 +2571,7 @@ bool NPC::Death(Mob* killer_mob, int32 damage, uint16 spell, EQ::skills::SkillTy
 
 		entity_list.RemoveFromAutoXTargets(this);
 
-		if (killer->GetUltimateOwner() && killer->GetUltimateOwner()->IsClient()) {
+		if (killer && killer != 0 && killer->GetUltimateOwner() && killer->GetUltimateOwner()->IsClient()) {
```

Comment:
Should be just `killer != nullptr` (yes I know the rest of the code is gross)

Reply:
So, do you want the line to be:

if (killer != nullptr && killer != 0 && killer->GetUltimateOwner() && killer->GetUltimateOwner()->IsClient()) {
if (killer && killer != nullptr && killer->GetUltimateOwner() && killer->GetUltimateOwner()->IsClient()) {
if (killer != nullptr && killer->GetUltimateOwner() && killer->GetUltimateOwner()->IsClient()) {

Reply:
Guys lets close the loop on this

Reply:
The last one, they all do the same and likely compile down to the same thing. Someday I'll tackle rewriting the section of code, it's ugly and likely doesn't need that many checks on the killer pointer, but that first if is too gross to unpack atm

Reply:
I will just add what was requested to what I already had.  It will be overkill, but I think it will get us beyond the issue tonight.

Reply:
I must have been typing out my response at the same time as you.  I will get your requested change in now.

Reply:
@mackal @Akkadius 
Latest commit reflects the requested changes.